### ISSUE TITLE
Podtrac Prepend no longer added to feed MP3s

### ISSUE THREAD

**Issue by mariahism** [a8cteam51/onbeing#68]
We had a prepend on each MP3 in the podcast feeds that would allow the podcast to be tracked by Podtrac. This would populate automatically when we entered the MP3. When we migrated to WordPress, this feature was lost, and consequently all the data we would have collected. It's a top priority to find a way to auto-insert the Podtrac URL back into the GUID on all podcast feeds. 

You can see that it's missing here: https://onbeing.org/podcasts/podcast.xml

And present here: http://test.onbeing.org/podcasts/podcast.xml 
<guid>
http://www.podtrac.com/pts/redirect.mp3/feeds.soundcloud.com/stream/269430313-samar-jarrah-wajahat-ali-sahar-ullah-et-al-revealing-ramadan-1.mp3
</guid>

**Comment by michiecat** [comment]
@mariahism My team and I are out this week so we won't be able to implement anything on the site until next week at the earliest (because no one will be around should there be any problems). However I can do some initial investigation to get the ball rolling. I have a few clarifying questions: 

1. To confirm, the last time you had Podtrac was when you were using Drupal? E.g. you've never had Podtrac when you were running on Pantheon either? 

2. Do you remember how Podtrac was implemented on the Drupal site? E.g. was it via a plugin, or something you manually added when creating each episode?

3. Do you still have login access to http://test.onbeing.org? If so, would you be able to give us access? Access to the code would be helpful so we can see how it was implemented before. 

4. If you have an account at Podtrac that you could add us to as well, that would be helpful for testing.

**Comment by tgilliss** [comment]
Hello. To answer your questions:

#1. Yes. The Drupal backend would insert the snippet "
http://dts.podtrac.com/redirect.mp3/" before "feeds.soundcloud..." but that
wasn't handled in our initial WP creation. (e.g. The download URL should
read like this:
http://dts.podtrac.com/redirect.mp3/feeds.soundcloud.com/stream/322104957-glenn-beck-what-you-do-will-be-a-pivot-point.mp3
)

#2. It was dynamically inserted, but I don't know by which method. We
didn't manually insert though.

#3. Our Pantheon instances are still live so you should be able to access.
Glad to walk you through it.

#4. It's not very useful, and we'll be glad to spot-check and measure once
a solution is implemented.

On this note, I'm wondering if for now there's someway to do a mass find
and replace for mp3 files on our episodes and then we'll manually update
them until we develop a plugin that accounts for this type of tracking
service. So, anything that begins with "http://feeds.soundcloud.com/..."
would be replaced with "
http://dts.podtrac.com/redirect.mp3/feeds.soundcloud.com/..." instead.
Thoughts?

If not, we'll manually update them episode by episode.

Thank you!
T

...................
*Trent Gilliss*
Publisher & Editor-in-Chief
*On Being <http://onbeing.org/> + **The Civil Conversations Project
<http://civilconversationsproject.org>*

651.206.2022 / @trentgilliss <http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

On Tue, May 16, 2017 at 5:57 PM, Michelle Langston <notifications@github.com

**Comment by michiecat** [comment]
I looked into a fix for this. I'm trying to do this programmatically by taking the URL of the soundcloud file and doing a regex find/replace on the beginning of the URL. I need to do some more testing and will report back once I have a fix ready for review on obtester.com.

**Comment by tgilliss** [comment]
Thank you! Let me know if I can be of use to you.
T

...................
*Trent Gilliss*
Publisher & Editor-in-Chief
*On Being <http://onbeing.org/> + **The Civil Conversations Project
<http://civilconversationsproject.org>*

651.206.2022 / @trentgilliss <http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

**Comment by michiecat** [comment]
@tgilliss I've got a fix for this, and will have it ready for you to test on http://obtester.com on Monday, June 26th. I'll notify you here when it's ready for you to test.

**Comment by tgilliss** [comment]
Thanks, Michelle. As an FYI, our team manually updated all of the URLs to
have this prepend. Please let me know if this might be something you can
strip with a script once the solution is ready or if I will have to ask my
team to reverse manually. (I'm sure hoping the former is possible!) ;)

Trent

...................
*Trent Gilliss*
Publisher & Editor-in-Chief
*On Being <http://onbeing.org/> + **The Civil Conversations Project
<http://civilconversationsproject.org>*

651.206.2022 / @trentgilliss <http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

On Sun, Jun 25, 2017 at 5:06 PM, Michelle Langston <notifications@github.com

**Comment by michiecat** [comment]
@tgilliss Ah, so what I can do is add a check in my solution to check if the link already has the prepend, and if it doesn't, add it. We can look into writing a script to strip the links from the existing ones later (it might be easier to do that with a MySQL find and replace, but we'd test it on obtester.first).

That said, caveat with my solution is that the prepend will only be added programmatically. That is, the podcast link that gets saved in the database won't contain the prepend. This way if you ever decide that the prepend is no longer needed, you won't need to go back and remove it. Is that what you prefer?

**Comment by michiecat** [comment]
@tgilliss My fix is now available on obtester.com. I'm going to test this a bit more, and if all checks out, we can push it next week, after the holiday. If a link doesn't have the prepend, it'll add it. Links that already have the prepend won't be affected.

**Comment by tgilliss** [comment]
Hi Michelle--

Just back from vacation and ready to jump in on this one. What would be the
best way to test this to make sure that it works properly and also doesn't
negatively affect any of our feeds or mobile app?

Warm regards,
Trent


...................
*Trent Gilliss*
Publisher & Editor-in-Chief
*On Being <http://onbeing.org/> + **The Civil Conversations Project
<http://civilconversationsproject.org>*

651.206.2022 / @trentgilliss <http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

On Thu, Jun 29, 2017 at 9:52 AM, Michelle Langston <notifications@github.com

**Comment by michiecat** [comment]
Hi @tgilliss! Welcome back. Right now, the fix is up on http://obtester.com/, the development site. To test, I suggest:

1. Log into http://obtester.com/wp-login.php (your login should be the same as on the main site). 
2. Open up a few podcast episodes and add the prepend to some of them (but not all of them). 
3. Open up the feeds (linked below) and check that:
 a ) The prepend shows up in the `guid` for all episodes
 b) For the episodes where you manually added the prepend in Step 2 above, the prepend looks OK (e.g. it's not added twice or has any other oddities). 
4. Run the test site's feeds through any other feed checkers that you use, including testing via iTunes: https://help.apple.com/itc/podcasts_connect/#/itcac471c970. 

When we're ready to make it live, we have several options: 

1. We can implement it on just one feed to start with, and if all goes well, roll out the fix to the other feeds. 
2. I'll be on hand to roll back any changes if there are problems (it's easy to roll back because this fix does not affect anything in the database). 

Feeds to test (these are all on the testing site):

https://obtester.com/feed/podcast/podcast_mobile
https://obtester.com/feed/podcast/podcast
https://obtester.com/feed/podcast/becoming-wise
https://obtester.com/feed/podcast/episodes_only
https://obtester.com/feed/podcast/extras_only

**Comment by tgilliss** [comment]
Hi Michelle--

Testing this out, it doesn't look like the prepend is being added to all
GUIDs that don't have it added to the admin interface. For instance, when I
search the podcast with Sherwin Nuland the Podtrac prepend isn't being
added. Can you verify this?

Thank you!
Trent

...................
*Trent Gilliss*
Editor-in-Chief / Publisher / Cofounder
*On Being <https://onbeing.org/> + On Being Studios <https://onbeing.org/>*

mobile 651.206.2022 / twitter @trentgilliss
<http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

On Mon, Jul 10, 2017 at 5:17 PM, Michelle Langston <notifications@github.com

**Comment by michiecat** [comment]
@tgilliss I've reapplied the fix for this on the dev site (I had temporarily removed it so that we could push other changes through first). You can now see it if you search for the podcast with Sherwin Nuland here: https://obtester.com/feed/podcast/podcast

I'd say we go ahead and push this fix if you're satisfied, and if there are problems, it's easy to revert (since nothing gets changed in the database). Let me know!

**Comment by tgilliss** [comment]
Hi Michelle--

After looking at the RSS feed, the GUIDs look correct but the enclosure
URLs remain the same and should display this prepend too. Here's an example:

<enclosure url="
http://feeds.soundcloud.com/stream/318532120-unedited-helen-fisher-this-is-your-brain-on-sex.mp3
" length="68254374"type="audio/mpeg"/>

...................
*Trent Gilliss*
Editor-in-Chief / Publisher / Cofounder
*On Being <https://onbeing.org/> + On Being Studios <https://onbeing.org/>*

mobile 651.206.2022 / twitter @trentgilliss
<http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

On Mon, Jul 31, 2017 at 4:27 PM, Michelle Langston <notifications@github.com

**Comment by ChrisHardie** [comment]
Hi @tgilliss. I updated our solution to make sure to take both the guid and the enclosure URLs into account. We'll now auto-replace any mention in any feed of a "feeds.soundcloud.com" link with a version that includes the podtrac prefix (and leave the link alone otherwise, including cases where you're already including the podtrac prefix). 

If that looks good to you in test feeds like 
https://obtester.com/feed/mobile/podcast
https://obtester.com/feed/podcast/podcast
then we can go ahead and update the live site.

Moving forward your editors will not need to manually add the podtrac prefix, but can also leave any existing cases where they did as-is. We can still look later at cleaning things up so that all of the links are consistent across episodes.

**Comment by tgilliss** [comment]
Chris, this looks good. Once you move the fix into production can you let
us know immediately so we can test and make sure all is working properly in
production?

Thank you!
Trent

...................
*Trent Gilliss*
Editor-in-Chief / Publisher / Cofounder
*On Being <https://onbeing.org/> + On Being Studios <https://onbeing.org/>*

mobile 651.206.2022 / twitter @trentgilliss
<http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

**Comment by michiecat** [comment]
@tgilliss We've pushed this into production. Please let me know if there are problems, and we can roll it back.

**Comment by tgilliss** [comment]
Noticing this issue was reopened. Should I hold off on verifying in
production?

...................
*Trent Gilliss*
Editor-in-Chief / Publisher / Cofounder
*On Being <https://onbeing.org/> + On Being Studios <https://onbeing.org/>*

mobile 651.206.2022 / twitter @trentgilliss
<http://twitter.com/trentgilliss>

1619 Hennepin Ave
Minneapolis, MN 55403

**Comment by michiecat** [comment]
@tgilliss I reopened it by accident when alerting you previously (I clicked "reopen and comment" before :D) . Please verify that all looks good in production on your end. I've just checked the feeds and it looks good from my end. If there are problems, we can re-open.
